### ISSUE TITLE
URL-based cachebuster can't be disabled

### ISSUE THREAD

**Issue by JaveleyQAQ** [PortSwigger/param-miner#94]
When I deselect 'params: query', the feature does not take effect, and the query parameters are still included in the request.

![](https://media.discordapp.net/attachments/1161955063531569183/1186247238540853319/123.gif?ex=66ac0615&is=66aab495&hm=8a1f426b0706251affdd96fe7679bd579ea6633b647384f1aba5a2530ce9ffa8&=&width=763&height=541)

**Comment by albinowax** [comment]
Sorry, I'm not sure I understand you. Can you let me know which type of attack you are running, provide a screenshot of your Param Miner settings, and a screenshot that shows the unexpected result?

**Comment by albinowax** [comment]
Unfortunately I can't load this link.

**Comment by JaveleyQAQ** [comment]
> Unfortunately I can't load this link
![123](https://github.com/user-attachments/assets/6b13bd49-7b47-4db7-952a-a11e20bf1a50)
When you cancel the request parameters, they still appear in the GET request

**Comment by JaveleyQAQ** [comment]
> Unfortunately I can't load this link.

When you cancel the request parameters, they still appear in the GET request.
![123](https://github.com/user-attachments/assets/9715b84b-d461-4ee0-97d3-1dcd7e5a6817)

**Comment by albinowax** [comment]
Ahh thanks. This is the intended behaviour. 'params: query' refers to whether it attempts to scan existing query parameters. The parameter that you are observing in the query is just a a cache-buster.

**Comment by JaveleyQAQ** [comment]
> Ahh thanks. This is the intended behaviour. 'params: query' refers to whether it attempts to scan existing query parameters. The parameter that you are observing in the query is just a a cache-buster.

In the latest version, I can't find this JSON param option. If I don't want a random parameter to be inserted into the query, how should you configure it?

**Comment by JaveleyQAQ** [comment]
'Lab: Exploiting a mass assignment vulnerability'  In this lab, I'm unable to complete it using this extension because the parameter in the query is not being parsed.

**Comment by albinowax** [comment]
Sorry, this isn't currently possible but I'll fix it for you

**Comment by albinowax** [comment]
The attached release should fix this if you uncheck the new `include query-param in cachebusters` checkbox, can you test it? https://github.com/PortSwigger/param-miner/releases/tag/v1.5

**Comment by JaveleyQAQ** [comment]
The issue has been addressed, however, I encountered an error during the vulnerability scan. 

`Using albinowaxUtils v1.4
This extension should be run on the latest version of Burp Suite. Using an older version of Burp may cause impaired functionality.
Loaded Param Miner v1.5
Updating active thread pool size to 8
Loop 0
Loop 1
Queued 1 attacks from 1 requests in 0 seconds
Unrecognised type: 6
Initiating json bruteforce on 0a67007904f03c3181089fbc00fa0054.web-security-academy.net
Identified parameter on 0a67007904f03c3181089fbc00fa0054.web-security-academy.net: chosen_products:[0]:quantity~1
Error in thread: class burp.JsonParamNameInsertionPoint cannot be cast to class burp.ParamNameInsertionPoint (burp.JsonParamNameInsertionPoint and burp.ParamNameInsertionPoint are in unnamed module of loader burp.Za99 @1bef10e3). See error pane for stack trace.
Identified parameter on 0a67007904f03c3181089fbc00fa0054.web-security-academy.net: chosen_products:[0]:product_id~1
Error in thread: class burp.JsonParamNameInsertionPoint cannot be cast to class burp.ParamNameInsertionPoint (burp.JsonParamNameInsertionPoint and burp.ParamNameInsertionPoint are in unnamed module of loader burp.Za99 @1bef10e3). See error pane for stack trace.
Identified parameter on 0a67007904f03c3181089fbc00fa0054.web-security-academy.net: chosen_discount
Error in thread: class burp.JsonParamNameInsertionPoint cannot be cast to class burp.ParamNameInsertionPoint (burp.JsonParamNameInsertionPoint and burp.ParamNameInsertionPoint are in unnamed module of loader burp.Za99 @1bef10e3). See error pane for stack trace.
Identified parameter on 0a67007904f03c3181089fbc00fa0054.web-security-academy.net: chosen_products
Error in thread: class burp.JsonParamNameInsertionPoint cannot be cast to class burp.ParamNameInsertionPoint (burp.JsonParamNameInsertionPoint and burp.ParamNameInsertionPoint are in unnamed module of loader burp.Za99 @1bef10e3). See error pane for stack trace.
Identified parameter on 0a67007904f03c3181089fbc00fa0054.web-security-academy.net: chosen_products:[0]
Error in thread: class burp.JsonParamNameInsertionPoint cannot be cast to class burp.ParamNameInsertionPoint (burp.JsonParamNameInsertionPoint and burp.ParamNameInsertionPoint are in unnamed module of loader burp.Za99 @1bef10e3). See error pane for stack trace.
Identified parameter on 0a67007904f03c3181089fbc00fa0054.web-security-academy.net: chosen_products:[0]:item_price
Error in thread: class burp.JsonParamNameInsertionPoint cannot be cast to class burp.ParamNameInsertionPoint (burp.JsonParamNameInsertionPoint and burp.ParamNameInsertionPoint are in unnamed module of loader burp.Za99 @1bef10e3). See error pane for stack trace.
`
